src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -18,69 +14,23 @@ internal class AwsClient : IAwsClient {
 		_stsClient = stsClient;
 	}
 
-	AwsRoleState IAwsClient.GetRoles( string encodedSaml ) {
-		var samlToken = ParseSamlToken( encodedSaml );
-		var roleNodes = samlToken.SelectNodes( "//*[@Name=\"https://aws.amazon.com/SAML/Attributes/Role\"]/*" );
-
-		var roles = new List<AwsRole>();
-
-		if( roleNodes is not null ) {
-			foreach( XmlElement roleNode in roleNodes ) {
-				// SAML has value: <principal-arn>, <role-arn>
-				// The last part of the role-arn is a human readable name
-				string[] nodeContents = roleNode.InnerText.Split( "," );
-
-				roles.Add( new AwsRole(
-					RoleName: nodeContents[1].Split( "/" )[1],
-					PrincipalArn: nodeContents[0],
-					RoleArn: nodeContents[1]
-					) );
-			}
-		} else {
-			throw new BmxException( "Failed to retrieve roles" );
-		}
-		return new AwsRoleState( roles, encodedSaml );
-	}
-
 	async Task<AwsCredentials> IAwsClient.GetTokensAsync(
-		AwsRoleState state,
-		string selectedRole,
-		int durationInMinutes ) {
-		var role = state.AwsRoles
-			.Find( role => string.Equals( role.RoleName, selectedRole, StringComparison.OrdinalIgnoreCase ) );
-
-		if( role is not null ) {
-
-			// Generate access keys valid for 1 hour (default)
-			var authResp = await _stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest() {
-				PrincipalArn = role.PrincipalArn,
-				RoleArn = role.RoleArn,
-				SAMLAssertion = state.SamlString,
-			} );
-
-			return new AwsCredentials(
-				SessionToken: authResp.Credentials.SessionToken,
-				AccessKeyId: authResp.Credentials.AccessKeyId,
-				SecretAccessKey: authResp.Credentials.SecretAccessKey,
-				Expiration: authResp.Credentials.Expiration.ToUniversalTime()
-			);
-		}
-		throw new BmxException( "Invalid role selection" );
-	}
-
-	private XmlDocument ParseSamlToken( string encodedSaml ) {
-		string[] samlStatements = encodedSaml.Split( ";" );
-
-		// Process the B64 Encoded SAML string to get valid XML doc
-		var samlString = new StringBuilder();
-		foreach( string inputValueString in samlStatements ) {
-			samlString.Append( WebUtility.HtmlDecode( inputValueString ) );
-		}
```

Comment:
The processing here isn't necessary.
When the SAML response string is extracted from the value attribute [here](https://github.com/Brightspace/bmx/pull/322/files#diff-8b1e253817f43a932b584d4f407e885d9f64a848f09317feed09e86d5c3aa1cdL181), the XML parser already unescaped all the XML escape sequences.
Also, the semicolon is part of an XML escape sequence, so there was no reason to split the string by semicolon even if we do need to unescape the XML string.